lib/bot.js

```diff
@@ -86,6 +92,28 @@ Bot.prototype.attachListeners = function() {
     }
   }.bind(this));
 
+  this.ircClient.on('join', function(channel, nick) {
+    if (this.options.slackOptions.joinNotices) {
+      this.sendToSlack(this.nickname, channel, nick + ' has joined the ' + channel + ' channel');
+    }
+  }.bind(this));
+
+  this.ircClient.on('part', function(channel, nick, arg) {
+    if (this.options.slackOptions.partNotices) {
+      this.sendToSlack(this.nickname, channel, nick + ' has left the ' + channel + ' channel');
+    }
+  }.bind(this));
+
+  this.ircClient.on('quit', function(nick, reason, channels, message) {
+    if (this.options.slackOptions.quitNotices) {
+      var index;
+      for (index = 0; index < this.channels.length; index++) {
```

Comment:
Shouldn't this iterate over `channels` instead of `this.channels`? Also it'd probably be cleaner to use a [forEach](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach).